hoodie-spark/src/main/scala/com/uber/hoodie/IncrementalRelation.scala

```diff
@@ -88,8 +88,12 @@ class IncrementalRelation(val sqlContext: SQLContext,
         .get, classOf[HoodieCommitMetadata])
       fileIdToFullPath ++= metadata.getFileIdAndFullPaths(basePath).toMap
     }
+    // unset the path filter, otherwise if end_instant_time < latest_instant_time, path filter
```

Comment:
Can you describe this a little more please ? By reading offhand, this is not clear.

Reply:
Missing context is in how the RO view is implemented. It sets a path filter to filter all files except latest.. if we don't unset here, filter still kicks in and filters out all files when dealing with older instant ranges